### PR TITLE
Add a new protocol

### LINKED ISSUE #10: TCP support
Would be awesome if this also supported TCP backends (ideally with optional TLS support)!

### PR DESCRIPTION
**Pull request by juliens**
### What does this PR do?
Do we really need further explanations?

### Motivation
Fix an old issue!

Fixes #10

### More
- [x] Added/updated tests
- [x] Added/updated documentation

### Additional Notes
Yeah man!

Co-authored-by: Gérald Croës <gerald@containo.us>

### CODE REVIEW

**Comment by lexicoder** [comment]
Awesome!!!

**Comment by gaui** [comment]
Wow awesome!

File: docs/content/routing/routers/index.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+          # ...
+       [entrypoint.web-secure]
+          # ...
+       [entrypoint.other]
+          # ...
+
+    [tcp.routers]
+       [tcp.routers.Router-1]
+          entryPoints = ["web-secure", "other"] # won't listen to entrypoint web
+          rule = "Host(traefik.io)"
```
Comment (bbinet) [inline docs/content/routing/routers/index.md]: I was wondering if it was not supposed to be:
```
rule = "HostSNI(`traefik.io`)"
```

Comment (geraldcroes) [reply docs/content/routing/routers/index.md]: You are correct! Nice catch 👍 
Would you like to PR the change? or shall I?

Comment (bbinet) [reply docs/content/routing/routers/index.md]: Please review PR #4592

**Comment by kiwi-dejero** [comment]
Oh Boy Oh Boy Oh Boy !!

**Comment by mariufa** [comment]
Has anyone got this to work with postgres for instance?

**Comment by llitfkitfk** [comment]
Great!!!

**Comment by benyanke** [comment]
AWESOME.

**Comment by shantanugadgil** [comment]
This is genuinely awesome work!!!! Great job!!! :1st_place_medal: :100:

**Comment by grubyhs** [comment]
How to use it to expose redis? We have a problem with that

**Pull request merged**